library/core/src/ops/try_trait.rs

```diff
@@ -0,0 +1,225 @@
+use crate::ops::ControlFlow;
+
+/// The trait used for a variety of operations related to short-circuits,
+/// such as the `?` operator, `try {}` blocks, and `try_*` methods.
+///
+/// # Using `Try` in Generic Code
```

Comment:
Might be worth adding another example detailing how to implement a custom `Try` type, to give an idea of the motivating usecases.

Reply:
More docs are definitely good, but I think I'll leave that for a future PR, if that's ok.